cmd/juju/application/refresher/refresher.go

```diff
@@ -200,6 +200,10 @@ func (d *localCharmRefresher) Refresh() (*CharmID, error) {
 		}
 		return &CharmID{
 			URL: addedURL,
+			Origin: corecharm.Origin{
```

Comment:
It's important to keep the original origin for the refresh command in general.  While this will technically work because it's the local provider. There is a more optimal long term change to help avoid bugs in the future.